packages/realm/src/app-services/SyncConfiguration.ts

```diff
@@ -113,3 +114,26 @@ export function toBindingSyncConfig(config: SyncConfiguration): binding.SyncConf
       : binding.SyncSessionStopPolicy.AfterChangesUploaded,
   };
 }
+
+/** @internal */
+function validatePartitionValue(partitionValue: unknown) {
+  if (partitionValue === undefined) {
```

Comment:
We are missing a few other invalid cases here.  I would try and mirror what was written before in c++ (https://github.com/realm/realm-js/blob/master/src/js_sync.hpp#L77-L104)
For instance, currently the types `boolean`, `Date`, `Object`, `Array`, etc would all pass this validation.  In this case, I would verify that the value is typeof `string`, `number`, `objectId` or `null`.  If it isn't one of those, then throw.